homework/src/index.js

```diff
@@ -1,47 +1,110 @@
-'use strict';
+"use strick";
+
+function main(url) {
+
+  const root = document.getElementById("root");
 
-{
   function fetchJSON(url, cb) {
     const xhr = new XMLHttpRequest();
     xhr.open('GET', url);
     xhr.responseType = 'json';
     xhr.onload = () => {
-      if (xhr.status < 400) {
+      if (xhr.status === 200) {
         cb(null, xhr.response);
       } else {
-        cb(new Error(`Network error: ${xhr.status} - ${xhr.statusText}`));
+        cb({ text: `Error: ${xhr.status} - ${xhr.statusText}`, id: "id", val: "error" });
```

Comment:
By convention, error information is reported through a standard JavaScript Error object (as the original code did). Why did you choose to change this?